src/library/sidebarmodel.cpp

```diff
@@ -610,3 +634,100 @@ void SidebarModel::slotFeatureSelect(LibraryFeature* pFeature,
+    QString childData;
+
+    if (index.internalPointer() == this) {
+        // root feature selected
+        featureTitle = m_sFeatures[index.row()]->title().toString();
+    } else {
+        // child item selected
+        TreeItem* pTreeItem = static_cast<TreeItem*>(index.internalPointer());
+        if (pTreeItem) {
+            LibraryFeature* pFeature = pTreeItem->feature();
+            if (pFeature) {
+                featureTitle = pFeature->title().toString();
+                childData = pTreeItem->getData().toString();
+            }
+        }
+    }
+
+    if (!featureTitle.isEmpty()) {
+        m_pConfig->setValue(
+                ConfigKey("[Library]", "LastSelectedFeature"),
+                featureTitle);
+        m_pConfig->setValue(
+                ConfigKey("[Library]", "LastSelectedChild"),
+                childData);
+    }
+}
+
+QModelIndex SidebarModel::restoreSavedSelection() {
+    if (!m_pConfig || m_sFeatures.isEmpty()) {
+        return QModelIndex();
+    }
+
+    QString savedFeatureTitle = m_pConfig->getValueString(
+            ConfigKey("[Library]", "LastSelectedFeature"));
+    QString savedChildData = m_pConfig->getValueString(
+            ConfigKey("[Library]", "LastSelectedChild"));
+
+    if (savedFeatureTitle.isEmpty()) {
+        return QModelIndex();
+    }
+
+    // find the feature by title
+    for (int i = 0; i < m_sFeatures.size(); ++i) {
+        if (m_sFeatures[i]->title().toString() == savedFeatureTitle) {
+            // if no child data, return the feature root
+            if (savedChildData.isEmpty()) {
+                return index(i, 0);
+            }
+
+            // search for matching child
+            QAbstractItemModel* childModel = m_sFeatures[i]->sidebarModel();
+            if (childModel) {
+                // recursively search for matching child item
+                std::function<QModelIndex(const QModelIndex&)> findChild;
+                findChild = [&](const QModelIndex& parent) -> QModelIndex {
+                    int rows = childModel->rowCount(parent);
+                    for (int row = 0; row < rows; ++row) {
+                        QModelIndex childIndex = childModel->index(row, 0, parent);
+                        if (childIndex.isValid()) {
+                            TreeItem* pTreeItem = static_cast<TreeItem*>(
+                                    childIndex.internalPointer());
+                            if (pTreeItem && 
+                                pTreeItem->getData().toString() == savedChildData) {
+                                // found it! translate to sidebar model index
+                                return this->index(row, 0, this->index(i, 0));
+                            }
+                            // recursively search children
+                            QModelIndex found = findChild(childIndex);
+                            if (found.isValid()) {
+                                return found;
+                            }
```

Comment:
No need to iterate over children manually, just use [`pChildModel::match()`](https://doc.qt.io/qt-6/qabstractitemmodel.html#match) and stop at first match.